packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js

```diff
@@ -54,6 +57,73 @@ const Columns = ({
     },
     [columns, setColumnsObject]
   );
+
+  const setColumnStatus = (id, isVisible) => {
+    let list = document.querySelectorAll(
+      `li.${blockClass}__draggable-handleHolder`
+    );
+    for (let i = 0; i < list.length; i++) {
+      if (columns[i].id.includes(id)) {
+        if (isVisible) {
+          list[i].setAttribute('selected', 'true');
+        } else {
+          list[i].setAttribute('selected', 'false');
+        }
+      }
+    }
+  };
+
+  const initializeList = () => {
+    for (let i = 0; i < columns.length; i++) {
+      setColumnStatus(columns[i].id, columns[i].isVisible);
+    }
+  };
+
+  const setSelectAllColumnClassName = () => {
+    if (getVisibleColumnsCount() > 0) {
+      return `${blockClass}__draggable-handleHolder -selected`;
+    }
+    return `${blockClass}__draggable-handleHolder`;
+  };
+
+  const setSelectAllColumnSelected = () => {
+    let target = document.querySelectorAll('#customize-columns-select-all');
+    if (getVisibleColumnsCount() > 0) {
+      target[0].setAttribute('selected', 'true');
+    } else {
+      target[0].setAttribute('selected', 'false');
+    }
+  };
+
+  const setSelectAllCheckBox = () => {
+    let checkboxes = document.querySelectorAll(
+      `.${blockClass}__customize-columns-modal .bx--checkbox`
+    );
+    let selectAll = false;
+    if (getVisibleColumnsCount() == columns.length) {
+      selectAll = false;
+      for (let i = 1; i < checkboxes.length; i++) {
+        checkboxes[i].checked = false;
+        checkBoxChangeHandler(columns[i - 1], selectAll);
+        setSelectAllColumnSelected();
+      }
+    } else {
+      selectAll = true;
+      for (let i = 1; i < checkboxes.length; i++) {
+        checkboxes[i].checked = true;
+        checkBoxChangeHandler(columns[i - 1], selectAll);
+        setSelectAllColumnSelected();
+      }
+    }
```

Comment:
@kai-liu 

There is duplicate code here. So I would try to abstract out what's repeated.